packages/contracts/src/manager/upgradeInitializers/DiamondInit.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {LibDiamond} from "../libraries/LibDiamond.sol";
+import {IDiamondLoupe} from "../interfaces/IDiamondLoupe.sol";
+import {IDiamondCut} from "../interfaces/IDiamondCut.sol";
+import {IERC173} from "../interfaces/IERC173.sol";
+import "@openzeppelin/contracts/interfaces/IERC165.sol";
+import {Modifiers, GOVERNANCE_TOKEN_MINTER_ROLE, PAUSER_ROLE, CREDIT_NFT_MANAGER_ROLE, STAKING_MANAGER_ROLE, INCENTIVE_MANAGER_ROLE, GOVERNANCE_TOKEN_MANAGER_ROLE} from "../libraries/LibAppStorage.sol";
+import {AccessControlStorage} from "../libraries/AccessControlStorage.sol";
+
+// It is expected that this contract is customized if you want to deploy your diamond
+// with data from a deployment script. Use the init function to initialize state variables
+// of your diamond. Add parameters to the init funciton if you need to.
+
+contract DiamondInit is Modifiers {
+    // You can add parameters to this function in order to pass in
+    // data to set your own state variables
+    function init(address _admin) external {
+        // adding ERC165 data
+        LibDiamond.DiamondStorage storage ds = LibDiamond.diamondStorage();
+        ds.supportedInterfaces[type(IERC165).interfaceId] = true;
+        ds.supportedInterfaces[type(IDiamondCut).interfaceId] = true;
+        ds.supportedInterfaces[type(IDiamondLoupe).interfaceId] = true;
+        ds.supportedInterfaces[type(IERC173).interfaceId] = true;
+
+        _grantRole(AccessControlStorage.DEFAULT_ADMIN_ROLE, _admin);
+        _grantRole(GOVERNANCE_TOKEN_MINTER_ROLE, _admin);
+        _grantRole(PAUSER_ROLE, _admin);
+        _grantRole(CREDIT_NFT_MANAGER_ROLE, _admin);
+        _grantRole(STAKING_MANAGER_ROLE, _admin);
+        _grantRole(INCENTIVE_MANAGER_ROLE, _admin);
+        _grantRole(GOVERNANCE_TOKEN_MANAGER_ROLE, address(this));
+
+        s.dollarTokenAddress = address(this);
```

Comment:
if the dollar is part of the diamond we should add :
- ds.supportedInterfaces[type(IERC20Ubiquity).interfaceId] = true;
- the dollar facet in the deployment script

Reply:
> if the dollar is part of the diamond

No, it was a simple example that we can also initialize the state variables declared in the AppStorage struct when initializing diamond

Reply:
> if the dollar is part of the diamond we should add :

It should be removed now